pkg/bootstrap/platform/gcp.go

```diff
@@ -52,8 +54,12 @@ var (
 	GCPQuotaProjectVar = env.RegisterStringVar("GCP_QUOTA_PROJECT", "", "Allows specification of a quota project to be used in requests to GCP APIs.").Get()
 )
 
+var isMetadataEndpointAccessible bool
+
 var (
-	shouldFillMetadata = metadata.OnGCE
+	shouldFillMetadata = func() bool {
+		return metadata.OnGCE() && isMetadataEndpointAccessibleFn()
```

Comment:
There seems to be some disconnect between this and DiscoverWithTimeout. DiscoverWithTimeout is only checking OnGCE - if we cannot actually access the metadata server should we still consider it "On GCP" or just use the generic (none) platform?

if we do want to consider it "On GCP" but then not "shouldFillMetadata", I feel like we need some comments linking these two and explaining the interaction, etc.

We may also want to align the timeouts. DiscoverWithTimeout has a 5s timeout on OnGCE, but here we have no timeout on OnGCE but a 10s timeout on isMetadataEndpointAccessibleFn

Reply:
>  if we cannot actually access the metadata server should we still consider it "On GCP" or just use the generic (none) platform?

I opted to keep GcpEnv (instead of none) just in case that at some point the environments are used for more then just metadata.

I added a comment in the code and aligned the timeouts